pkg/graveler/ref/manager.go

```diff
@@ -230,80 +233,62 @@ func (m *Manager) DeleteBranch(ctx context.Context, repositoryID graveler.Reposi
-		return m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-			var commitID graveler.CommitID
-			err := tx.Get(&commitID, `SELECT commit_id FROM graveler_tags WHERE repository_id = $1 AND id = $2`,
-				repositoryID, tagID)
-			if err != nil {
-				return nil, err
-			}
-			return &commitID, nil
-		}, db.ReadOnly())
-	}))
-	if errors.Is(err, db.ErrNotFound) {
-		return nil, graveler.ErrTagNotFound
-	}
+func (m *KVManager) GetTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID) (*graveler.CommitID, error) {
+	tagKey := graveler.TagPath(tagID.String())
+	t := graveler.TagData{}
+	_, err := m.kvStore.GetMsg(ctx, graveler.TagPartition(repositoryID.String()), tagKey, &t)
 	if err != nil {
-		return nil, err
+		if errors.Is(err, kv.ErrNotFound) {
+			err = graveler.ErrTagNotFound
+		}
+		return nil, fmt.Errorf("%s: %w", tagKey, err)
 	}
-	return commitID.(*graveler.CommitID), nil
+	commitID := graveler.CommitID(t.CommitId)
+	return &commitID, nil
 }
 
-func (m *Manager) CreateTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID, commitID graveler.CommitID) error {
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		res, err := tx.Exec(`INSERT INTO graveler_tags (repository_id, id, commit_id) VALUES ($1, $2, $3)
-			ON CONFLICT DO NOTHING`,
-			repositoryID, tagID, commitID)
-		if err != nil {
-			return nil, err
-		}
-		if res.RowsAffected() == 0 {
-			return nil, graveler.ErrTagAlreadyExists
+func (m *KVManager) CreateTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID, commitID graveler.CommitID) error {
+	t := &graveler.TagData{
+		Id:       tagID.String(),
+		CommitId: commitID.String(),
+	}
+	tagKey := graveler.TagPath(tagID.String())
+	err := m.kvStore.SetMsgIf(ctx, graveler.TagPartition(repositoryID.String()), tagKey, t, nil)
+	if err != nil {
+		if errors.Is(err, kv.ErrPredicateFailed) {
+			err = graveler.ErrTagAlreadyExists
 		}
-		return nil, nil
-	})
+		return fmt.Errorf("save tag (tagKey %s): %w", tagKey, err)
+	}
 	return err
 }
 
-func (m *Manager) DeleteTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID) error {
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		r, err := tx.Exec(
-			`DELETE FROM graveler_tags WHERE repository_id = $1 AND id = $2`,
-			repositoryID, tagID)
-		if err != nil {
-			return nil, err
-		}
-		if r.RowsAffected() == 0 {
-			return nil, graveler.ErrNotFound
-		}
-		return nil, nil
-	})
-	if errors.Is(err, db.ErrNotFound) {
-		return graveler.ErrTagNotFound
+func (m *KVManager) DeleteTag(ctx context.Context, repositoryID graveler.RepositoryID, tagID graveler.TagID) error {
+	tagKey := graveler.TagPath(tagID.String())
+	err := m.kvStore.DeleteMsg(ctx, graveler.TagPartition(repositoryID.String()), tagKey)
+	if err != nil {
+		return fmt.Errorf("delete tag (tagKey %s): %w", tagKey, err)
```

Comment:
That's a behavior change from the original graveler, in case the key does not exist - returning the wrapped kv error, rather than `graveler.ErrNotFound`
Is that intentional?

Reply:
Related to #3640 ?

Reply:
yes